src/App.tsx

```diff
@@ -21,14 +24,18 @@ export const App: React.FC = () => {
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loader && <Loader />}
+              <TodoList
+                loader={loader}
+                setLoader={setLoader}
+                setModal={setModal}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {modal && <TodoModal setModal={setModal} />}
```

Comment:
you can just check if currentTodo exists
```suggestion
      {currentTodo && <TodoModal />}
```